### ISSUE TITLE
Field is being Disposed Analyzer

### ISSUE THREAD

**Issue by ckuhn203** [Vannevelj/VSDiagnostics#385]
I ran across some truly awful code today and believe it'd be a good analyzer to create. 

    private DbContext _context;

    // .... Lots of code...

    public void SomeMethod()
    {
        using (_context = new DbContext())
        {
            //...
        }
    }

This was a pattern in this particular code base, but it's dangerous, because it would be easy to forget that the field needs to be initialized on every call to it. An "ObjectDisposed" exception is just waiting to happen here. 

Possibly a quick fix could also be implemented that turns it into a local variable?

    public void SomeMethod()
    {
        using (var context = new DbContext())
        {
            //...
        }
    }

**Comment by Vannevelj** [comment]
Looks like a good analyzer to me. Let's add it in

**Comment by ckuhn203** [comment]
How could it have any relevant state info if it's being created in the `using` block?

**Comment by Vannevelj** [comment]
Yeah you're right, I don't know where I was going with that thought. I blame the late hour

**Comment by Vannevelj** [comment]
I'm thinking about this scenario again. Can you clarify what the problem is exactly? You can't assume that any `private` field is not null. Or at least, you shouldn't. What does this scenario do that isn't covered by that assumption?

**Comment by ckuhn203** [comment]
It's just a scope limiting issue. It makes no sense for it to even be a field if you're disposing of it in a block. The block should completely own the lifetime of the disposable.

**Comment by Vannevelj** [comment]
Ah okay, that clears it up (again).
